src/renderer/base/renderer.cpp

```diff
@@ -390,36 +387,26 @@ bool Renderer::_CheckViewportAndScroll()
     SMALL_RECT const srOldViewport = _viewport.ToInclusive();
     SMALL_RECT const srNewViewport = _pData->GetViewport().ToInclusive();
 
-    COORD coordDelta;
-    coordDelta.X = srOldViewport.Left - srNewViewport.Left;
-    coordDelta.Y = srOldViewport.Top - srNewViewport.Top;
-
-    for (auto engine : _rgpEngines)
+    if (srOldViewport == srNewViewport)

```

Comment:
If we check this early we can pull the rest of the function together into a single loop.